client/state/data-layer/wpcom-api-middleware.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Internal dependencies
+ */
+import handlers from './wpcom';
+
+/**
+ * WPCOM Middleware API
+ *
+ * Intercepts actions requesting data provided by the
+ * WordPress.com API and passes them off to the
+ * appropriate handler.
+ */
+export const middleware = store => next => action =>
+	// we won't use has( handlers, action.type )
+	// here becuase of performance implications
```

Comment:
also, if we're that worried about perf, maybe back this with a real `Map` instead of an object pretending to be a `Map`? It would mean we'd have to roll our own `mergeWith`, but that's not too hard, and lookup perf on a `Map` is much better than an `object`.